### PR TITLE
[Team6] Agent Patch -1

### PR DESCRIPTION
**Pull request by dejay-vu**
# Summary

Fix negative numbers in IIGO

## Additional Information

Auxiliary information goes here.

## Test Plan

- Add information on how you tested your changes.

### CODE REVIEW

File: internal/clients/team6/iigo.go [hunk 1]
```diff
@@ -58,17 +58,17 @@ func (c *client) CommonPoolResourceRequest() shared.Resources {
 	cprLeft := c.ServerReadHandle.GetGameState().CommonPool
 	reqResource := shared.Resources(3.0 * livingCost)
 
-	defer c.Logf("Request %v from common pool", reqResource)
-	//when common pool does not have enough resource, will not request
-
-	if ourStatus == shared.Critical {
-		return minThreshold - ourResources
+	if ourStatus == shared.Critical && ourResources < minThreshold {
+		return minThreshold - ourResources + shared.Resources(0.01)
```
Comment (lhl2617) [inline internal/clients/team6/iigo.go]: why 0.01? 0 is fine isnt it

Comment (dejay-vu) [reply internal/clients/team6/iigo.go]: so == minimumThreshold will not count as critical status right?

Comment (lhl2617) [reply internal/clients/team6/iigo.go]: ah. I see what you did there,
https://github.com/SOMAS2020/SOMAS2020/blob/73ed812b5add9f854209267fa1d66285f8a1199a/internal/server/helper.go#L63

you dont need that extra 0.01 to get out of critical

Comment (dejay-vu) [reply internal/clients/team6/iigo.go]: thx for this!

**Review by lhl2617 — APPROVED** [review]
image

**Pull request merged**